docs/book/src/developer/providers/v1.0-to-v1.1.md

```diff
@@ -0,0 +1,21 @@
+# Cluster API v1.0 compared to v1.1
+
+## Minimum Go version
+
+- The Go version used by Cluster API is still Go 1.16+
+
+## Controller Runtime version
+
+- The Controller Runtime version is still v0.10.+
```

Comment:
Is it possible we'll change this version in the 1.1. release?

Reply:
Yes it's very likely. But I would try to keep the doc sync with the current version of main and then adapt it accordingly (i.e. update the doc in the PRs where we make the corresponding changes, in that case upgrade CR)

Reply:
This is a great idea if we can enforce doc changes with changes like this. Is there any other category of changes we should enforce doc changes for? Some ideas:

1. Dependency changes 
2.  deprecations/removals 
3. API type introductions 
4. Contract changes

Reply:
I think in general that sums it up. Enforce more or less comes down to the contributor/reviewer to remember it :)

Reply:
We could add a small checklist to the [`PULL_REQUEST_TEMPLATE.md`](https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/PULL_REQUEST_TEMPLATE.md), this way the author gets reminded that they need to add docs for their change.

Reply:
We could add something in general about docs (and specifically about migration docs). 

I'm not 100% sure if we should though, given that the longer the template gets, the less folks read it.

But feel free to open a PR to amend the template!